samples/csharp_dotnetcore/21.corebot-app-insights/Dialogs/MainDialog.cs

```diff
@@ -2,100 +2,152 @@
 // Licensed under the MIT License.
 
 using System;
+using System.Collections.Generic;
+using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.Bot.Builder;
 using Microsoft.Bot.Builder.Dialogs;
-using Microsoft.Extensions.Configuration;
+using Microsoft.Bot.Schema;
 using Microsoft.Extensions.Logging;
 using Microsoft.Recognizers.Text.DataTypes.TimexExpression;
 
 namespace Microsoft.BotBuilderSamples.Dialogs
 {
     public class MainDialog : ComponentDialog
     {
-        protected readonly IConfiguration Configuration;
+        private readonly FlightBookingRecognizer _luisRecognizer;
         protected readonly ILogger Logger;
 
-        public MainDialog(IConfiguration configuration, ILogger<MainDialog> logger, IBotTelemetryClient telemetryClient)
+        // Dependency injection uses this constructor to instantiate MainDialog
+        public MainDialog(FlightBookingRecognizer luisRecognizer, BookingDialog bookingDialog, IBotTelemetryClient telemetryClient, ILogger<MainDialog> logger)
             : base(nameof(MainDialog))
         {
-            Configuration = configuration;
+            _luisRecognizer = luisRecognizer;
             Logger = logger;
-            TelemetryClient = telemetryClient;
 
-            AddDialog(new TextPrompt(nameof(TextPrompt))
-            {
-                TelemetryClient = telemetryClient,
-            });
-            AddDialog(new BookingDialog(telemetryClient));
+            this.TelemetryClient = telemetryClient;
+
+            // The TelemetryClient property for each added dialog will be set to
```

Comment:
Suggestions for comment:
* Move it above the line where you set the telemetryClient.
* Simplify the wording, perhaps something like:

>  // Set the telemetry client for this and all child dialogs.

Reply:
Done. Thanks for the suggestion.